### DISCUSSION TITLE
can't return regex as string due to serialization

### DISCUSSION THREAD

**Discussion by Chathula** [graphql/graphql-js#3237] [Q&A] [answered]
`\d{3}(-?|\s?)\d{3}`

I tried to return the above regex. but it always replaces `\` with `\\`. is there any way to prevent that? in the server console, it prints out the correct response. but when I check it using GraphiQL. it doesn't show what I expected. I save this value in DB as a regex. and then when returning I do this. `new RegExp(/\d{3}(-?|\s?)\d{3}/).toString()`

any solution for this?

**Comment by IvanGoncharov** [comment, marked as answer]
@Chathula Both JSON and Node's `inspect` add slash but it doesn't exist in string itself and you can check this using `console.log`:
```
new RegExp(/\d{3}(-?|\s?)\d{3}/).toString()
// '/\\d{3}(-?|\\s?)\\d{3}/'

console.log(new RegExp(/\d{3}(-?|\s?)\d{3}/).toString())
// /\d{3}(-?|\s?)\d{3}/
```
